code/_core/obj/item/clothing/belt/bandoliers/_bandoliers.dm

```diff
@@ -0,0 +1,149 @@
+		.["stored_bullets"][k] = v
+
+/obj/item/clothing/belt/bandoliers/load_item_data_pre(var/mob/living/advanced/player/P,var/list/object_data,var/loadout=FALSE)
+	RUN_PARENT_SAFE
+	for(var/k in object_data["stored_bullets"])
+		var/v = object_data["stored_bullets"][k]
+		stored_bullets[text2path_safe(k)] = v
+
+/obj/item/clothing/belt/bandoliers/Finalize()
+	. = ..()
+	update_bullet_count()
+	update_sprite()
+
+/obj/item/clothing/belt/bandoliers/update_icon()
+	. = ..()
+	icon_state_worn = initial(icon_state_worn)
+	if(num_overlays == 0)
+		if(bullet_count)
+			icon_state_worn = "[icon_state_worn]_filled"
+			icon_state = "full"
+		else
+			icon_state = "inventory"
+	else
+		if(bullet_count > 0)
+			icon_state_worn = "[icon_state_worn]_filled"
+
+/obj/item/clothing/belt/bandoliers/update_overlays()
+	. = ..()
+	if(num_overlays > 0)
+		var/step = max_bullets/num_overlays
+		if(bullet_count)
+			var/image/I = new/image(icon,"bullet_[CEILING(bullet_count/step,1)]")
+			add_overlay(I)
+
+/obj/item/clothing/belt/bandoliers/clicked_on_by_object(var/mob/caller,var/atom/object,location,control,params)
+
+	if(istype(object,bullet_type))
+		INTERACT_CHECK
+		INTERACT_CHECK_OBJECT
+		INTERACT_DELAY(2)
+
+		var/obj/item/bullet_cartridge/C = object
+
+		if(caller.attack_flags & CONTROL_MOD_DISARM)
+			if(!stored_bullets[C.type])
+				caller.to_chat(span("warning","There are no charges of that type left in \the [src.name]!"))
+				return TRUE
+			var/amount_added = C.add_item_count(1)
+			if(amount_added)
+				stored_bullets[C.type] -= amount_added
+				bullet_count -= amount_added
+				caller.to_chat(span("notice","You add a charge to your hand. There are [stored_bullets[C.type]] charges of that type left."))
+				if(stored_bullets[C.type] <= 0)
+					stored_bullets -= C.type
+				update_sprite()
+		else
+			var/amount_added = -C.add_item_count(-min(C.amount,max(0,max_bullets - bullet_count)))
+			if(amount_added)
+				if(!stored_bullets[C.type])
+					stored_bullets[C.type] = amount_added
+				else
+					stored_bullets[C.type] += amount_added
+				update_bullet_count()
+				if(bullet_count > max_bullets) //Failsafe in case someone somehow adds more bullets to the bandolier than the bandolier can hold
+					C.add_item_count(bullet_count - max_bullets)
+					stored_bullets[C.type] -= (bullet_count - max_bullets)
+					update_bullet_count()
+				update_sprite()
+		return TRUE
+
+	if(istype(object,/obj/item/bullet_cartridge)&&!istype(object,stored_bullets))
+		INTERACT_CHECK
+		INTERACT_CHECK_OBJECT
+		INTERACT_DELAY(2)
+		caller.to_chat(span("warning","This type of bullet won't fit into the [src.name]!"))
+		return TRUE
+
+	if(istype(object,/obj/hud/inventory))
+		INTERACT_CHECK
+		INTERACT_CHECK_OBJECT
+		INTERACT_DELAY(2)
+
+		if(!length(stored_bullets))
+			caller.to_chat(span("warning","There are no bullets left!"))
+			return TRUE
+
+		var/obj/hud/inventory/I = object
+		var/obj/item/bullet_cartridge/C = pickweight(stored_bullets)
+		if(!C)
```

Comment:
pickweight(stored_bullets) shouldn't return null as there is a list check above. has it returned null during testing?

Reply:
I honestly don't know how or why that phrase got in there, never experienced any issues with pickweight(stored_bullets) though.